publisher/upload/upload.go

```diff
@@ -0,0 +1,399 @@
+package upload
+
+import (
+	"fmt"
+	"github.com/newrelic/infrastructure-publish-action/publisher/config"
+	"github.com/newrelic/infrastructure-publish-action/publisher/download"
+	"github.com/newrelic/infrastructure-publish-action/publisher/lock"
+	"github.com/newrelic/infrastructure-publish-action/publisher/utils"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"time"
+)
+
+const (
+	//FileTypes
+	typeFile         = "file"
+	typeZypp         = "zypp"
+	typeYum          = "yum"
+	typeApt          = "apt"
+	repodataRpmPath  = "/repodata/repomd.xml"
+	signatureRpmPath = "/repodata/repomd.xml.asc"
+	aptPoolMain      = "pool/main/"
+	aptDists         = "dists/"
+	commandTimeout   = time.Hour * 1
+)
+
+var (
+	l = log.New(log.Writer(), "", 0)
+)
+
+func uploadArtifact(conf config.Config, schema config.UploadArtifactSchema, arch string, upload config.Upload) (err error) {
```

Comment:
since we are splitting things, would make sense to place each upload type in a different file?

Reply:
what do you mean?